### ISSUE TITLE
MSVC: no import library for laszip_api3.dll generated

### ISSUE THREAD

**Issue by chringer** [LASzip/LASzip#45]
If I generate a Visual Studio 2017 project using CMake and build the project, there are only laszip3.lib and laszip3.dll as well as laszip_api3.dll generated, but the import lib laszip_api3.lib is missing.

**Comment by rapidlasso** [comment]
I'm just linking my executables directly with the laszip_api.c code snippet. Did you try that?

**Comment by hobu** [comment]
It might be that the cmake needs to be updated to create the export library. I had to major-version the DLL/LIB combo to prevent mixing with laszip.dll based on the older version(s) of the library.

**Comment by chringer** [comment]
Hi guys,

thank you for your fast reply and sorry for my late answer.

> linking my executables directly with the laszip_api.c code snippet
What does it mean to link an exectable to a code snippet? Do you mean building the snippet as part of my own executable?

> It might be that the cmake needs to be updated
I'm using the (nearly) latest version 3.11.0... or what do you mean? I'm getting the same result when building a clean check-out of this github repository.

Thank you very much for your help!

**Comment by rapidlasso** [comment]
> Do you mean building the snippet as part of my own executable?

Yes. This is how I use the DLL for testing and [in my example](https://github.com/LAStools/LAStools/tree/master/LASzip/example).

**Comment by chringer** [comment]
Yeah, this works but it isn't really the way how one should use an external library as fas as I know... Linking against a (perhaps pre-built) library would be better than requiring the source code of this lib, I think. Thank you anyway.

**Comment by chringer** [comment]
I've one update on this issue: The import lib laszip_api3.lib is only missing if I build in _Debug_ configuration, in _Release_ configuration it is generated as expected.

**Comment by hobu** [comment]
I'm sorry that I'm not familiar who Debug vs. Release would cause DLL export issues.

**Comment by chringer** [comment]
For guys having the same problem: the following does the trick. Simply insert the following piece of code into the file `LASzip\dll\CMakeLists.txt`, just right before the call to `LASZIP_ADD_LIBRARY`:

`if(NOT LASZIP_BUILD_STATIC)
    add_definitions(-DLASZIP_DYN_LINK)
    add_definitions(-DLASZIP_SOURCE)
endif()`
